docs/api-documentation.md

```diff
@@ -0,0 +1,370 @@
+# API Documentation
+
+This document provides a brief overview of the different API endpoints, their usage and purpose.
+
+## Authentication
+
+This section lists the authentication related requests such as login.
+
+### Login
+
+Allows the user to login into the portal and receive the auth token from the API. Requires the user email and password values.
+
+**User Login Flow**
+
+Here is a step-by-step breakdown of the user login flow:
+
+1. **Client-Side Request:**
```

Comment:
All of these can be headings. Please remove the colon character.

Reply:
I tried the ##### heading but doesn't look good 
![image](https://github.com/OsmosysSoftware/dotnet-foundation/assets/121787291/50b592d5-b788-4d6c-aa5d-cb512e40929d)

The bold one look better 
![image](https://github.com/OsmosysSoftware/dotnet-foundation/assets/121787291/143f58a6-b806-48af-9f9d-a4d8e83650df)